### DISCUSSION TITLE
Improve the visual display of function parameter lists (toggle display of type hints)

### DISCUSSION THREAD

**Discussion by willingc** [microsoft/pylance-release#2673] [Enhancement]
Historically, type hints for type checking parameters in a method/function were not used. We are seeing more and more use of type  hints for static and dynamic checking today.

Readability of these functions becomes more difficult when the types are displayed especially as the list of parameters increases. If there was a way to toggle on and off the display of these types, it would improve readability and cognitive understanding of a page of code. The type hints would remain part of the source code but would be optional to display.

cc @brettcannon @tonybaloney

**Comment by tonybaloney** [comment]
<img width="1209" alt="screenshot 2021-12-15 at 07 58 09" src="https://user-images.githubusercontent.com/1532417/146078301-2b297dcc-b2b1-446d-8f02-29b08a34ab4a.png">
Hi @willingc , just to clarify, you're referring to the window that appears when you hover over a function or method (as shown in the screenshot)?

**Comment by willingc** [comment]
@tonybaloney That would be an awesome step in  the right direction. Ideally a toggle on the whole file, display or not.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
